llvm/lib/Analysis/MemoryBuiltins.cpp

```diff
@@ -729,13 +742,16 @@ SizeOffsetType ObjectSizeOffsetVisitor::compute(Value *V) {
           SOT.second.getBitWidth() > 1 ? SOT.second + Offset : SOT.second};
 }
 
-SizeOffsetType ObjectSizeOffsetVisitor::computeImpl(Value *V) {
+SizeOffsetType ObjectSizeOffsetVisitor::computeValue(Value *V) {
   if (Instruction *I = dyn_cast<Instruction>(V)) {
     // If we have already seen this instruction, bail out. Cycles can happen in
     // unreachable code after constant propagation.
     auto P = SeenInsts.try_emplace(I, unknown());
     if (!P.second)
       return P.first->second;
+    ++InstructionsVisited;
+    if (InstructionsVisited > ObjectSizeOffsetVisitorMaxVisitInstructions)
```

Comment:
Could check `SeenInsts.count()` instead of a separate variable here?

Reply:
`SeenInsts` is a cache that can carry over between calls to `ObjectSizeOffsetVisitor::compute()`, but I think we want `ObjectSizeOffsetVisitorMaxVisitInstructions` to apply per call to `ObjectSizeOffsetVisitor::compute()`. This does lead us to a weird situation where successive calls to `compute()` will return different answers since we may explore more from cached results, but I think that's better than if somebody reuses a `ObjectSizeOffsetVisitor` and later calls to it all return `unknown()` since previous calls filled up the cache to `ObjectSizeOffsetVisitorMaxVisitInstructions` size

Reply:
Good point. I was not aware that ObjectSizeOffsetVisitor is designed for reuse, but apparently AddressSanitizer does use it that way.

Possibly we should be clearing SeenInsts for each call as well...